azurerm/internal/services/keyvault/key_vault_certificate_resource.go

```diff
@@ -127,6 +142,12 @@ func resourceKeyVaultCertificate() *schema.Resource {
 										Type:     schema.TypeString,
 										Required: true,
 										ForceNew: true,
+										ValidateFunc: validation.StringInSlice([]string{
+											string(keyvault.EC),
+											string(keyvault.ECHSM),
+											string(keyvault.RSA),
+											string(keyvault.RSAHSM),
+										}, false),
```

Comment:
We should be case-insensitive here until v3.0 to avoid breaking existing configs

```suggestion
										}, true), // TODO: make this case-sensitive in v3.0
```